bokehjs/src/coffee/models/annotations/label_set.coffee

```diff
@@ -22,13 +22,25 @@ export class LabelSetView extends TextAnnotationView
       @connect(@model.change, () ->
         @set_data(@model.source)
         @render())
+      @connect(@model.source.streaming, () ->
+        @set_data(@model.source)
+        @render())
+      @connect(@model.source.patching, () ->
+        @set_data(@model.source)
+        @render())
       @connect(@model.source.change, () ->
         @set_data(@model.source)
         @render())
     else
       @connect(@model.change, () ->
         @set_data(@model.source)
         @plot_view.request_render())
+      @connect(@model.source.streaming, () ->
+        @set_data(@model.source)
+        @plot_view.request_render())
+      @connect(@model.source.patching, () ->
+        @set_data(@model.source)
+        @plot_view.request_render())
```

Comment:
Alternatively can add a separate `set_data_and_render` function and keep `set_data` as it is. Could do that in response to @canavandl comment above too.